hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieParquetDataBlock.java

```diff
@@ -162,8 +163,33 @@ protected <T> ClosableIterator<HoodieRecord<T>> readRecordsFromBlockPayload(Hood
     return iterator;
   }
 
+  @Override
+  protected <T> ClosableIterator<T> readRecordsFromBlockPayload(HoodieReaderContext<T> readerContext) throws IOException {
+    HoodieLogBlockContentLocation blockContentLoc = getBlockContentLocation().get();
+
+    // NOTE: It's important to extend Hadoop configuration here to make sure configuration
+    //       is appropriately carried over
+    Configuration inlineConf = FSUtils.buildInlineConf(blockContentLoc.getHadoopConf());
```

Comment:
But the problem is that it will do `new Configuration` per block even though `hadoopConf` is at file reader level. Can we avoid new-ing configuration per block?